src/ipahealthcheck/ipa/certs.py

```diff
@@ -583,6 +584,125 @@ def check(self):
+                             msg='Unable to load CA cert: {error}')
+                return False
+            if cacert != dbcacert:
+                yield Result(plugin, constants.ERROR,
+                             msg=('CA Certificate from %s does not match %s'
+                                  % (dbpath, paths.IPA_CA_CRT)))
+                return False
+            return True
+
+        try:
+            cacert = x509.load_certificate_from_file(paths.IPA_CA_CRT)
+        except Exception as e:
+            yield Result(self, constants.ERROR,
+                         error=str(e),
+                         msg='Unable to load CA cert: {error}')
+            return
+
+        # Ensure CA cert entry from LDAP matches /etc/ipa/ca.crt
+        try:
+            dn = DN('cn=%s IPA CA' % api.env.realm,
+                    'cn=certificates,cn=ipa,cn=etc',
+                    api.env.basedn)
+            entry = self.conn.get_entry(dn)
+        except errors.NotFound:
+            yield Result(self, constants.ERROR,
+                         msg='CA Certificate entry not found in LDAP')
+            return
+        if cacert != entry['CACertificate'][0]:
+            yield Result(self, constants.ERROR,
+                         msg=('CA Certificate entry in LDAP does not '
+                              'match %s' % paths.IPA_CA_CRT))
+            return
+
+        # Ensure NSS DBs have matching CA certs for /etc/ipa/ca.crt
+        serverid = realm_to_serverid(api.env.realm)
+        dspath = paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % serverid
+        cacertds_ok = yield from match_cacert_and_db(self, cacert, dspath)
+        cacertnss_ok = yield from match_cacert_and_db(self, cacert,
+                                                      paths.IPA_NSSDB_DIR)
+
+        if cacertds_ok and cacertnss_ok:
+            yield Result(self, constants.SUCCESS)
+
+
+@registry
+class IPADogtagCertsMatchCheck(IPAPlugin):
+    """
+    Check if dogtag certs present in both NSS DB and LDAP match
+    """
+    requires = ('dirsrv',)
+
+    @duration
+    def check(self):
+        if not self.ca.is_configured():
+            logger.debug('CA is not configured, skipping connectivity check')
+            return
+
+        def match_ldap_nss_cert(plugin, ldap, db, cert_dn, attr, cert_nick):
+            try:
+                entry = ldap.get_entry(cert_dn)
+            except errors.NotFound:
+                yield Result(plugin, constants.ERROR,
+                             msg='%s entry not found in LDAP' % cert_dn)
+                return False
+            try:
+                nsscert = db.get_cert_from_db(cert_nick)
+            except Exception as e:
+                yield Result(plugin, constants.ERROR,
+                             error=str(e),
+                             msg=('Unable to load %s certificate:'
+                                  '{error}' % cert_nick))
+                return False
+            ldapcert = entry[attr][0]
```

Comment:
It's possible that there could be older, expired/revoked certificates in userCertificate. It would be safer to walk all the certs looking for a match.

It might also be more robust to ensure they are in the same format. For example load them as x509 objects and compare them rather than comparing the raw blobs.